x/superfluid/spec/12_minting.md

```diff
@@ -0,0 +1,31 @@
+<!--
+order: 12
+-->
+
+# Minting
+
+
+Superfluid module has the ability to arbitrarily mint and burn Osmo through the `bank` module. This is potentially dangerous so we strictly constrain it's ability to do so.
+This authority is mediated through the `mintOsmoTokensAndDelegate` and `forceUndelegateAndBurnOsmoTokens` keeper methods, which are in turn called by message handlers (`SuperfluidDelegate`, `SuperfluidUndelegate`, and `SuperfluidDelegateMore`) as well as by `RefreshIntermediaryDelegationAmounts` as part of an `Epoch` hook.
```

Comment:
Is there an invariant that makes sense to add to the mint coins itself? (Using the MintCoinsRestrictionFn)

Reply:
I think it makes more sense to check invariants at the end of a tx / epoch, and revert/panic if the check fails, since it's a restriction on the state of the account as a whole rather than each call to mint/burn.

I guess as an extra level of safety, we could have a MintCoinsRestrictionFn which says the amount to mint + already delegated <= expectedDelegation?